api/src/test/java/org/openmrs/obs/BinaryDataHandlerTest.java

```diff
@@ -9,42 +9,149 @@
  */
 package org.openmrs.obs;
 
-import static org.junit.Assert.assertArrayEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.any;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
 
+import java.io.File;
+
+import org.apache.commons.lang3.RandomUtils;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.openmrs.Obs;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.obs.handler.AbstractHandler;
 import org.openmrs.obs.handler.BinaryDataHandler;
+import org.openmrs.util.OpenmrsUtil;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
 
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ AbstractHandler.class, OpenmrsUtil.class, Context.class })
 
 public class BinaryDataHandlerTest {
-
-    @Test
-    public void shouldReturnSupportedViews() {
-        BinaryDataHandler handler = new BinaryDataHandler();
-        String[] actualViews = handler.getSupportedViews();
-        String[] expectedViews = { ComplexObsHandler.RAW_VIEW };
-
-        assertArrayEquals(actualViews, expectedViews);
-    }
-
-    @Test
-    public void shouldSupportRawView() {
-        BinaryDataHandler handler = new BinaryDataHandler();
```

Comment:
Did you intentionally remove these?

Reply:
They aren't removed (all of the cases you commented as far as I can see). I'm not sure if it has to do with Eclipse's Auto Formatting (I took the OpenMRS Profile for it) or with me moving stuff. Just check the right side of the diff and you will see they should all still be there, just moved. I haven't changed them at all.

Reply:
It takes more time to review such a pull request because one needs to differentiate between what was actually removed (red) and what was added back (green). How about if you just added your changes at the very end of the file?

Reply:
I do understand your point of course. However, adding the @Before method before the @Test methods for ordering reasons is pretty understandable I think. I can't imagine other developers add code to files "chronologically"? The same goes for the new variables I introduced to the test classes, including the @Mock variable for the administration service. Why would I add class variables below?

Reply:
In that case, you can start by just adding them at the end such that the pull request does not show the other methods as moved. After it is merged, you can create a new pull request for simply formatting purposes.

Reply:
What about imports and other declarations that go before the class, do they cause the same problems? I can't move them down.

Reply:
imports and variable declarations are fine.

Reply:
I will do that and see if the diffs come out clearer, hopefully I can manage to do it later today. Meanwhile, are the null errors gone now? And could you take a look at my last comment concerning the codacy remarks?